compiler/passes/buildDefaultFunctions.cpp

```diff
@@ -1827,10 +1899,17 @@ void buildDefaultDestructor(AggregateType* ct) {
 
     fn->retType = dtVoid;
 
+    if (ct->isUnion()) {
+      fn->insertAtTail(buildResetUnionField(fn->_this, ct, NULL));
+    }
+
     fn->insertAtTail(new CallExpr(PRIM_RETURN, gVoid));
 
     ct->symbol->defPoint->insertBefore(new DefExpr(fn));
 
+    if (ct->isUnion()) {
+      normalize(fn);
+    }
```

Comment:
Why would we only want to normalize if the new function was a union?  Shouldn't we just always normalize the functions we create in buildDefaultFunctions?

Reply:
I don't want to change behavior for non-unions at this time. I believe that buildDefaultDestructor didn't run normalize on the function because it is so simple (consisting only of a PRIM_RETURN) that it is already normalized.

In other words - I added more code for unions in the conditional above - and that code needs to be normalized.